### ISSUE TITLE
Box.copy() returns a dictionary

### ISSUE THREAD

**Issue by rshap91** [cdgriffith/Box#28]
Is this intentional?

```
d = dict.fromkeys([1,2,3,4],0)
d = box.Box(d)
print(type(d))
print(type(d.copy()))
---------------------------
>> <class 'box.Box'>
>> <class 'dict'>
```

**Comment by cdgriffith** [comment]
It has not come up yet, but I agree that it would make more sense to have the copy return a Box object as well. Will look into deepcopy's behavior as well.

**Comment by cdgriffith** [comment]
Fixed in 3.1 https://github.com/cdgriffith/Box/commit/252e2f574213708cbbcd42bdec442434384127fe
